src/EntityId/EntityIdPager.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Wikibase\DataModel\Services\EntityId;
+
+use Wikibase\DataModel\Entity\EntityId;
+
+/**
+ * A cursor for paging through EntityIds.
+ *
+ * @since 3.7
+ *
+ * @license GPL-2.0+
+ * @author Daniel Kinzler
+ */
+interface EntityIdPager {
```

Comment:
Might be worth changing so it implements the Fetcher interface from https://github.com/JeroenDeDauw/BatchingIterator, which then allows for having an Iterator and hiding that a fetcher is used